Automattic/wp-calypso
# Pages: Show pages in hierarchical order

Before:
![image](https://cloud.githubusercontent.com/assets/363749/25497699/0e027a5c-2b4b-11e7-9526-ec41143e6b93.png)

After:
![image](https://cloud.githubusercontent.com/assets/363749/25497652/ee4643c4-2b4a-11e7-8d58-4ae57c2e968a.png)

To test:
* Verify that pages that are children of another page are shown hierarchically when there are fewer than 100 pages
* Verify that drafts, scheduled, and trashed show up as normal, non-hierarchically.
* Verify that search shows up as normal, non-hierarchically.
* Verify that orphaned children show up as top-level pages
* Verify that `menu_order` is obeyed for sorting (after hierarchy)

Notes:
* This PR only tackles displaying pages hierarchically -- the design will be tackled in a future PR
* This PR also, for the moment, removes the BlogPostsPage block. It will be added back in a modified form in a future PR.
